src/App.xaml.cs

```diff
@@ -176,6 +183,12 @@ protected async override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventA
 
     private void OnActivated(object? sender, AppActivationArguments args)
     {
+        if (args.Kind == ExtendedActivationKind.ToastNotification)

```

Comment:
I think we can leverage `IActivationHandler` here by adding a new activation handler for the type [ToastNotificationActivatedEventArgs](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.activation.toastnotificationactivatedeventargs?view=winrt-22621). 

 Check ProtocolActivationHandler.cs for an example.